### PR TITLE
FOGL-8115: Add Control Entry-points API Flow

### PR DESCRIPTION
**Pull request by monika-Dianomic**
Form validation related changes and other css fixes

### CODE REVIEW

File: src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.html [hunk 1]
```diff
@@ -419,7 +416,7 @@
                     <div class="field">
                         <div class="columns">
                             <div class="column is-2">
-                                <div *ngIf="editMode && allowExecute" class="field has-addons is-fullwidth">
+                                <div *ngIf="editMode && af.permitted" class="field has-addons is-fullwidth">
```
Comment (praveen-garg) [inline src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.html:419]: All conditionals stuff with edit mode for permitted needs proper testing; for each role based user; and not auth mode.

File: src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.html [hunk 2]
```diff
@@ -450,7 +447,7 @@
                                     </p>
                                     <p class="control">
                                     <button (click)="saveAPIFlow(apiFlowForm.value)" type="submit" class="button is-small is-link"
-                                        [disabled]="apiFlowForm?.invalid">Save</button>
+                                    [disabled]="apiFlowForm?.invalid || (selectedType === 'operation' && apiFlowForm.controls.operation_name.value === '')">Save</button>
```
Comment (praveen-garg) [inline src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.html:450]: check == '' after trim?

File: src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.ts [hunk 3]
```diff
@@ -170,6 +174,11 @@ export class AddEditAPIFlowComponent implements OnInit {
             this.editMode = true;
             this.ngProgress.done();
             this.af = data;
+            
+            this.apiFlowForm.get('name').setValue(data.name);
+            this.apiFlowForm.get('description').setValue(data.description);
+            this.apiFlowForm.get('operation_name').setValue(data.operation_name);
```
Comment (praveen-garg) [inline src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.ts:180]: what if operation name is not there; and its of type Write?

**Review by praveen-garg — COMMENTED** [review]
Override variables modal has different modal body appearance, from list vs details page; Also contents are not vertically centre.

File: src/styles.css [hunk 4]
```diff
@@ -404,6 +404,10 @@ a.dropdown-item:hover {
     font-size: 0.70rem;
   }
 
+  .bi-icon:hover {
```
Comment (praveen-garg) [inline src/styles.css:407]: remove this too.

File: src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.ts [hunk 5]
```diff
@@ -194,8 +201,11 @@ export class AddEditAPIFlowComponent implements OnInit {
 
     saveAPIFlow(data) {
       let payload = this.af;
-
+      
+      payload.name = data.name;
+      payload.description = data.description;
       payload.type = this.selectedType;
+      payload.operation_name = data.operation_name;
```
Comment (praveen-garg) [inline src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.ts:208]: if exists or always?

File: src/app/components/core/control-dispatcher/api-flow/api-flow.component.html [hunk 6]
```diff
@@ -93,37 +92,42 @@
...
-                                        <div class="column is-3">
-                                            <div class="control">
-                                                <label class="label label-name has-text-left">{{v.value.vName}}</label>
-                                            </div>
-                                        </div>
-                                        <div class="column is-4">
-                                            <div class="control">
-                                                <input formControlName="vValue" placeholder="Variable value" class="input is-small" name="value" type="text">
+        <div class="columns">
+            <div class="column is-3"></div>
```
Comment (hemantgarg923) [inline src/app/components/core/control-dispatcher/api-flow/api-flow.component.html:96]: Please make it `column is-1`

File: src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.html [hunk 7]
```diff
@@ -477,7 +474,7 @@
 <section class="modal-card-body">
     <div class="columns">
         <div class="column is-3"></div>
```
Comment (hemantgarg923) [inline src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.html:476]: Please make it `column is-1`

Comment (hemantgarg923) [inline src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.html:476]: Please make it `column is-1`

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**